src/settings.js

```diff
@@ -1,45 +1,46 @@
 import { isPlainObject } from './helpers';
 import { Value } from './values';
 
-export default class Settings {
-  static prefix;
+function parseValue(composedSettings, value, envName, envPrefix) {
+  if (typeof value === 'function') {
+    value = value.apply(composedSettings);
+  }
+
+  if (value instanceof Value) {
+    if (value.options.envName === undefined) {
+      value.options.envName = envName;
+    }
+    if (value.options.envPrefix === undefined) {
+      value.options.envPrefix = envPrefix;
+    }
+    value = value.value;
+  } else if (isPlainObject(value)) {
+    Object.entries(value).forEach(([k, v]) => {
+      value[k] = parseValue(composedSettings, v);
+    });
+  }
+
+  return value;
+}
 
-  constructor(settings) {
-    this.cache = new Map();
-    this.proxy = new Proxy(settings, {
-      get: (target, key) => {
-        if (this.cache.has(key)) {
-          return this.cache.get(key);
+export function composeSettings(settings, prefix) {
```

Comment:
I usually prefer multiline parameters, like `settings`, to bet he last parameter. It would read better to me as `composeSettings(prefix, settings)`. I guess that doesn't allow prefix to be optional though.

Reply:
yeah I do too. i wish JS had real named arguments so life could just be better.